packages/react-components/tests/robots/robot-marker.spec.tsx

```diff
@@ -0,0 +1,71 @@
+import { act, render as render_, RenderResult } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+import { ColorContext, ColorManager, RobotMarker } from '../../lib';
+import { makeRobot } from './test-utils';
+
+describe('robot-markers', () => {
+  let colorManager: ColorManager;
+  let handler: { onClick: () => void; robotColor: () => Promise<string> };
+
+  beforeEach(() => {
+    // TextEncoder is not available in node
+    colorManager = new ColorManager();
+    handler = {
+      onClick: () => console.log('mock'),
+      robotColor: async () => 'black',
+    };
+
+    spyOn(handler, 'onClick');
+    spyOn(handler, 'robotColor');
+
+    colorManager.robotPrimaryColor = handler.robotColor;
```

Comment:
Since we are running the tests in browser, this comment shouldn't be valid anymore. There also shouldn't be a need to mock `ColorManager` now.

Reply:
Had a lot of errors without instantiating ColorManager. 

```
ERROR in [at-loader] ./tests/robots/robot-marker.spec.tsx:23:32 
    TS2739: Type 'Spy<Func>' is missing the following properties from type 'ColorManager': conflictHighlight, robotPrimaryColor, robotColorFromCache, _robotColorFromId, _robotColorCache

ERROR in [at-loader] ./tests/robots/robot-marker.spec.tsx:32:32 
    TS2322: Type 'Spy<Func>' is not assignable to type 'ColorManager'.
```
Also, I think we need that for this test to work:

```
colorManager.robotPrimaryColor = jasmine.CreateSpy();

  it("uses ColorManager to determine robot's color", async () => {
    await render(<RobotMarker robot={makeRobot()} fleetName="test_fleet" footprint={1} />);
    expect(colorManager.robotPrimaryColor).toHaveBeenCalled();
  });
```